features/FEATURE_LWIP/lwip-interface/lwip-eth/arch/TARGET_Freescale/k64f_emac.c

```diff
@@ -526,15 +524,14 @@ static err_t k64f_low_level_output(struct netif *netif, struct pbuf *p)
 
   /* Wait until a descriptor is available for the transfer. */
   /* THIS WILL BLOCK UNTIL THERE ARE A DESCRIPTOR AVAILABLE */
-  while (g_handle.txBdCurrent->control & ENET_BUFFDESCRIPTOR_TX_READY_MASK)
-    osSemaphoreWait(k64f_enet->xTXDCountSem.id, osWaitForever);
+  osSemaphoreWait(k64f_enet->xTXDCountSem.id, osWaitForever);
```

Comment:
Blocking is well defined at the Network socket API layer, so that is another option. If it was done there, then the LWIP thread would be free to continue processing RX packets.